fdbserver/ptxn/test/TestSerialization.cpp

```diff
@@ -139,3 +248,228 @@ TEST_CASE("/fdbserver/ptxn/test/ProxyTLogPushMessageSerializer") {
 
 	return Void();
 }
+
+namespace {
+
+// Serialize a bunch of {version, subsequence, mutation}
+void serializeVersionedSubsequencedMutations(
+    ptxn::TLogStorageServerMessageSerializer& serializer,
+    const std::vector<ptxn::VersionSubsequenceMutation> versionSubsequenceMutations) {
+	Version currentVersion = invalidVersion;
+	for (const auto& item : versionSubsequenceMutations) {
+		if (currentVersion != item.version) {
+			if (currentVersion != invalidVersion) {
+				serializer.completeVersionWriting();
+			}
+			currentVersion = item.version;
+			serializer.startVersionWriting(currentVersion);
+		}
+		serializer.writeSubsequenceMutationRef(item.subsequence, item.mutation);
+	}
+
+	if (currentVersion != invalidVersion) {
+		serializer.completeVersionWriting();
+	}
+
+	// Intended to leave the serializer open to new versions
+}
+
+} // anonymous namespace
+
+TEST_CASE("/fdbserver/ptxn/test/TLogStorageServerMessageSerializer/basic") {
```

Comment:
Nit: this and other `TEST_CASE` below does not call `wait()`, so I guess you can change each function as a C++ function that returns a bool value. Then just create one `TEST_CASE` that assert each of the C++ function returns true.